protean/config/auto_link.hpp

```diff
@@ -7,70 +7,59 @@
 #define PROTEAN_CONFIG_AUTO_LINK_HPP
 
 #ifndef PROTEAN_VERSION_HPP
-#  include <protean/version.hpp>
+#include <protean/version.hpp>
 #endif
 
 #ifndef PROTEAN_LIB_NAME
-#   error "Macro PROTEAN_LIB_NAME not set (internal error)"
+	#error "Macro PROTEAN_LIB_NAME not set (internal error)"
 #endif
 
-#if defined(_MSC_VER)
+#ifdef _MSC_VER
 
-#if (_MSC_VER <= 1400)
-#   define PROTEAN_LIB_TOOLSET "vc80"
-#elif (_MSC_VER == 1500)
-#   define PROTEAN_LIB_TOOLSET "vc90"
-#elif (_MSC_VER == 1600)
-#   define PROTEAN_LIB_TOOLSET "vc100"
-#elif (_MSC_VER == 1700)
-#   define PROTEAN_LIB_TOOLSET "vc110"
-#elif (_MSC_VER == 1800)
-#   define PROTEAN_LIB_TOOLSET "vc120"
-#elif (_MSC_VER == 1900)
-#   define PROTEAN_LIB_TOOLSET "vc140"
-#else
-#   error "Unhandled _MSC_VER"
-#endif

```

Comment:
MS have changed their policy for `_MSC_VER` bumps in VS2017, so it's necessary to check as below now.